File: src/main/java/com/gpuflight/agent/model/LogWrapper.java
package com.gpuflight.agent.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public record LogWrapper(
    @JsonProperty("agentSendingTime") long agentSendingTime,
    @JsonProperty("data") String data,
    @JsonProperty("type") String type,
    @JsonProperty("hostname") String hostname,
    @JsonProperty("ipAddr") String ipAddr
) {}

File: src/main/kotlin/com/gpuflight/agent/model/LogWrapper.kt
package com.gpuflight.com.gpuflight.agent.model

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
data class LogWrapper(
    val agentSendingTime: Long,
    val data: String,
    val type: String,
    val hostname: String,
    val ipAddr: String
)
